chat-client/src/client/mynahUi.ts

```diff
@@ -777,11 +775,16 @@ export const createMynahUi = (
         }
     }
 
-    const sendToPrompt = (params: SendToPromptParams) => {
+    const sendToPrompt = (params: SendToPromptParams & { autoSubmit?: boolean; prompt: ChatPrompt }) => {
         const tabId = getOrCreateTabId()
         if (!tabId) return
 
-        mynahUi.addToUserPrompt(tabId, params.selection, 'code')
+        if (params.autoSubmit) {
+            messager.onChatPrompt({ prompt: params.prompt, tabId, context: undefined }, 'contextMenu')
+            initializeChatResponse(mynahUi, tabId, params.prompt.prompt)
```

Comment:
I wasn't able to figure out how to have the code selection auto submit, so I just sent mine along with the prompt

Reply:
The problem is that for explain code one prompt is sent to the user and one is sent to the backend. In the vscode implementation we're sending prompt to the user and escaped prompt to the backend

Reply:
whats the difference between the two prompt fields?

Reply:
escapedPrompt typically just trims the start/end of the string AFAICT

Reply:
is there any reason to directly use `params.prompt.prompt` then?

Reply:
Oh, I see this is because we are using fundamentally different values here: https://github.com/aws/aws-toolkit-vscode/pull/7194/files, I'm assuming this is the first time we want to split where we send the prompt. Yeah, I don't see a better way to handle this, but maybe worth building this support into the chat client more naturally.

Reply:
It's basically a hack, we're showing `params.prompt.prompt` to the user and then agentic chat controller sends `params.prompt.escapedPrompt` to the backend. The prompt that goes to the user is just: "Explain the issue with x,y,z" and the prompt that goes to the backend sends like 60 extra lines to the backend with a bunch more context. That's why its seperated